internal/v2/s2a_v2_e2e_test.go

```diff
@@ -0,0 +1,158 @@
+		t.Errorf("net.Listen(tcp, :0) failed: %v", err)
+	}
+	s := grpc.NewServer()
+	s2av2pb.RegisterS2AServiceServer(s, &fakes2av2.Server{})
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Errorf("s.Serve(%v) failed: %v", lis, err)
+		}
+	}()
+	return lis.Addr().String()
+}
+
+func startFakeS2AOnUDS(t *testing.T) string {
+	dir, err := ioutil.TempDir("/tmp", "socket_dir")
+	if err != nil {
+		t.Errorf("unable to create temporary directory: %v", err)
+	}
+	udsAddress := filepath.Join(dir, "socket")
+	lis, err := net.Listen("unix", filepath.Join(dir, "socket"))
+	if err != nil {
+		t.Errorf("net.Listen(unix, %s) failed: %v", udsAddress, err)
+	}
+	s := grpc.NewServer()
+	s2av2pb.RegisterS2AServiceServer(s, &fakes2av2.Server{})
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Errorf("s.Serve(%v) failed: %v", lis, err)
+		}
+	}()
+	return fmt.Sprintf("unix://%s", lis.Addr().String())
+}
+
+// startServer starts up a server and returns the address that it is listening
+// on.
+func startServer(t *testing.T, s2aAddress string) string {
+	creds, err := NewServerCreds(s2aAddress)
+	if err != nil {
+		t.Errorf("NewServerCreds(%s) failed: %v", s2aAddress, err)
+	}
+
+	lis, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Errorf("net.Listen(tcp, :0) failed: %v", err)
+	}
+	s := grpc.NewServer(grpc.Creds(creds))
+	helloworldpb.RegisterGreeterServer(s, &server{})
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Errorf("s.Serve(%v) failed: %v", lis, err)
+		}
+	}()
+	return lis.Addr().String()
+}
+
+// runClient starts up a client and calls the server.
+func runClient(ctx context.Context, t *testing.T, clientS2AAddress, serverAddr string) {
+	creds, err := NewClientCreds(clientS2AAddress)
+	if err != nil {
+		t.Errorf("NewClientCreds(%s) failed: %v", clientS2AAddress, err)
+	}
+	dialOptions := []grpc.DialOption{
+		grpc.WithTransportCredentials(creds),
+		grpc.WithBlock(),
+	}
+
+	grpclog.Info("client dialing server at address: %v", serverAddr)
+	// Establish a connection to the server.
+	conn, err := grpc.Dial(serverAddr, dialOptions...)
+	if err != nil {
+		t.Errorf("grpc.Dial(%v, %v) failed: %v", serverAddr, dialOptions, err)
+	}
+	defer conn.Close()
+
+	// Contact the server.
+	c := helloworldpb.NewGreeterClient(conn)
+	req := &helloworldpb.HelloRequest{Name: clientMessage}
+	grpclog.Infof("client calling SayHello with request: %v", req)
+	resp, err := c.SayHello(ctx, req, grpc.WaitForReady(true))
+	if err != nil {
+		t.Errorf("c.SayHello(%v, %v) failed: %v", ctx, req, err)
+	}
+	if got, want := resp.GetMessage(), "Hello "+clientMessage; got != want {
+		t.Errorf("r.GetMessage() = %v, want %v", got, want)
+	}
+	grpclog.Infof("client recieved message from server: %s", resp.GetMessage())
+}
+
+func TestEndToEndUsingFakeS2AOverTCP(t *testing.T) {
+	// TODO(rmehta19): Set S2A_ACCESS_TOKEN env variable once #46 merges.
+	// Start the fake S2As for the client and server.
+	serverHandshakerAddr := startFakeS2A(t)
```

Comment:
For all the variables that look like "serverHandshakeAddr" or similar for a client, can we rename them to "serverS2AAddr"?

The "Handshaker" term is very v1-focused, and no longer really applies for S2Av2.